packages/@lwc/babel-plugin-component/src/post-process/dedupe-imports.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+function defaultImport(t, specifiers) {
+    const defaultImport = specifiers.find((s) => t.isImportDefaultSpecifier(s));
+    return defaultImport && defaultImport.local.name;
+}
+
+module.exports = function dedupeImports({ types: t }, path) {
+    const body = path.get('body');
+    const importStatements = body.filter((s) => s.isImportDeclaration());
+    const visited = new Map();
+
+    importStatements.forEach((importPath) => {
+        const sourceLiteral = importPath.node.source;
+        // If we have seen the same source, we will try to dedupe it
+        if (visited.has(sourceLiteral.value)) {
+            const visitedImport = visited.get(sourceLiteral.value);
+            const visitedSpecifiers = visitedImport.node.specifiers;
+            const visitedDefaultImport = defaultImport(t, visitedSpecifiers);
+
+            // We merge all the named imports unless is a default with the same name
+            let canImportBeRemoved = true;
+            importPath.node.specifiers.forEach((s) => {
+                if (visitedDefaultImport && t.isImportDefaultSpecifier(s)) {
+                    if (visitedDefaultImport !== s.local.name) {
+                        canImportBeRemoved = false;
+                    }
```

Comment:
Please add a test for this if it doesn't already exist

Reply:
@ekashida look at the tests I had to fix :) we have already 30 tests that are testing this

Reply:
Sure...I saw that you added this code after updating all the tests so I thought it might have been something we need to add a new test case for.